lib/active_merchant/billing/gateways/decidir_plus.rb

```diff
@@ -0,0 +1,151 @@
+module ActiveMerchant #:nodoc:
+  module Billing #:nodoc:
+    class DecidirPlusGateway < Gateway
+      self.test_url = 'https://developers.decidir.com/api/v2'
+      self.live_url = 'https://live.decidir.com/api/v2'
+
+      self.supported_countries = ['ARG']
+      self.default_currency = 'ARS'
+      self.supported_cardtypes = %i[visa master american_express discover]
+
+      self.homepage_url = 'http://decidir.com.ar/home'
+      self.display_name = 'Decidir Plus'
+
+      def initialize(options = {})
+        requires!(options, :public_key, :private_key)
+        super
+      end
+
+      def purchase(money, payment, options = {})
+        post = {}
+        post[:site_transaction_id] = options[:site_transaction_id]
+        post[:token] = options[:payment_id].split('|')[0]
+        post[:payment_method_id] = 1
+        post[:bin] = payment.number.to_s[0..5]
+        post[:amount] = money
+        post[:currency] = options[:currency] || self.default_currency
+        post[:installments] = options[:installments] || 1
+        post[:payment_type] = options[:payment_type] || 'single'
+        post[:sub_payments] = []
+
+        commit(:post, 'payments', post)
+      end
+
+      def refund(money, authorization, options = {})
+        post = {}
+        post[:amount] = money
+
+        commit(:post, "payments/#{add_reference(authorization)}/refunds")
+      end
+
+      def store(payment, options = {})
+        post = {}
+        add_payment(post, payment, options)
+
+        commit(:post, 'tokens', post)
+      end
+
+      def supports_scrubbing?
+        true
+      end
+
+      def scrub(transcript)
+        transcript.
+          gsub(%r((Apikey: )\w+), '\1[FILTERED]').
+          gsub(%r(("card_number\\?"\s*:\s*\\?")[^"]*)i, '\1[FILTERED]').
+          gsub(%r(("security_code\\?"\s*:\s*\\?")[^"]*)i, '\1[FILTERED]')
+      end
+
+      private
+
+      def add_reference(authorization)
+        return unless authorization
+
+        authorization.split('|')[0]
+      end
+
+      def add_invoice(post, money, options)
+        post[:amount] = amount(money)
```

Comment:
Bringing this up since it’s happened a few times lately, does the gateway expect the amount to always be an integer? If so, it may be worthwhile to throw in a `.to_i` now and avoid the errors that might pop up in core integration tests later on.

Reply:
I'm actually not calling this method, so I just removed it. The `amount` method converts the integer representation of the amount to a 2 decimal string representation of the amount, but Decidir+ is just looking for that integer representation.